gallery/src/main/java/com/redmadrobot/gallery/ui/GalleryDialogFragment.kt

```diff
@@ -0,0 +1,173 @@
+
+        draggableView.setOnDragListener { offset ->
+            rootView.alpha = 1 - min(Math.abs(offset / (3 * dismissPathLength)), 1f)
+            viewPager.translationY = -offset
+        }
+        draggableView.setOnReleaseDragListener { offset ->
+            if (Math.abs(offset) > dismissPathLength) {
+                viewPager.isVisible = false
+                dismiss()
+            } else {
+                rootView.alpha = 1f
+                viewPager.translationY = 0f
+            }
+        }
+
+        lastPosition = savedInstanceState?.getInt(STATE_LAST_CHECKED_ITEM_INDEX) ?: position
+
+        mediaViewController = MediaViewController(
+                viewPager = viewPager,
+                onCurrentItemChangeListener = { index ->
+                    lastPosition = index
+                },
+                onPlayerControllerVisibilityListener = {},
+                onImageZoomListener = { isZoomed -> draggableView.isDraggingEnabled = !isZoomed }
+        ).apply {
+            bind(listOfMedia)
+            setCurrentItemIndex(lastPosition)
+        }
+    }
+
+    @CallSuper
+    override fun onResume() {
+        activity?.run {
+            initialOrientation = requestedOrientation
+            if (isRotationEnabled) {
+                requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR
+            }
+        }
+        applyWindowFullscreenStyle()
+        super.onResume()
+    }
+
+    @CallSuper
+    override fun onPause() {
+        activity?.requestedOrientation = initialOrientation
+        super.onPause()
+    }
+
+    @CallSuper
+    override fun onSaveInstanceState(outState: Bundle) {
+        outState.putInt(STATE_LAST_CHECKED_ITEM_INDEX, lastPosition)
+        super.onSaveInstanceState(outState)
+    }
+
+    @CallSuper
+    override fun onDestroyView() {
+        super.onDestroyView()
+        mediaViewController.release()
+    }
+
+    private fun applyWindowFullscreenStyle() {
+        dialog?.window?.apply {
+            attributes = attributes?.apply {
+                width = WindowManager.LayoutParams.MATCH_PARENT
+                height = WindowManager.LayoutParams.MATCH_PARENT
+            }
+        }
+    }
+
+    companion object {
+        private const val STATE_LAST_CHECKED_ITEM_INDEX = "state_last_checked_item_index"
+        private const val ARG_LIST_OF_MEDIA = "arg_list_of_media"
+        private const val ARG_INITIALLY_CHECKED_ITEM_INDEX = "arg_initially_checked_item_index"
+        private const val ARG_FORCE_ROTATION = "arg_force_rotation"
+        private const val ARG_LAYOUT_RES = "arg_layout_res"
+        private const val ARG_DRAG_LAYOUT_ID = "arg_drag_layout_id"
+        private const val ARG_VIEW_PAGER_ID = "arg_view_pager_id"
+        private const val ARG_SWIPE_TO_DISMISS_LENGTH_DIMEN = "arg_swipe_to_dismiss_length_dimen"
+
+        @JvmOverloads
+        @JvmStatic
+        fun <T : GalleryDialogFragment> T.applyGalleryParams(
```

Comment:
Create this method without default parameters, please!
But add this method:
```kotlin
fun create(
  media: ArrayList<Media>,
  position: Int,
  isRotationEnabled: Boolean = false
) : GalleryDialogFragment =
 GalleryDialogFragment().applyGalleryParams(...) //with all default parameters
```

Reply:
Done